src/PrintEncrypted.cpp

```diff
@@ -0,0 +1,240 @@
+/* A simple program that demonstrates a simple homomorphic program on 4 input values.
+ * */
+
+#include <iostream>
+#include <NTL/ZZ.h>
+#include <NTL/BasicThreadPool.h>
+#include "FHE.h"
+#include "timing.h"
+#include "EncryptedArray.h"
+#include <NTL/lzz_pXFactoring.h>
+
+#include <cassert>
+#include <cstdio>
+
+namespace std {} using namespace std;
+
+int main(int argc, char *argv[])
+{
+  ArgMapping amap;
+
+  // Technical parameter. Use if you want to explicitly set generator elements
+  // for the plaintext arrays.
+  Vec<long> gens;
+  amap.arg("gens", gens, "use specified vector of generators", NULL);
+  amap.note("e.g., gens='[562 1871 751]'");
+
+  // Technical parameter. The order of each generator specified in ords.
+  Vec<long> ords;
+  amap.arg("ords", ords, "use specified vector of orders", NULL);
+  amap.note("e.g., ords='[4 2 -4]', negative means 'bad'");
+
+  // Random seed used in scheme.
+  long seed=0;
+  amap.arg("seed", seed, "PRG seed");
+
+  // The number of rounds of encrypted computation. If R > 1, then we need to
+  // "bootstrap" between rounds which adds a heavy computational overhead.
+  long R=1;
```

Comment:
now i'm really gunna complain...

Reply:
:-). Will try to add in descriptive names for these one-char variable names instead.